src/style/colors.ts

```diff
@@ -199,10 +199,58 @@ export class Colors {
 
     const sliced = tints.slice(0, 8);
     const adjusted = adjustSaturation(sliced, color);
-    const palette = adjusted || sliced;
+    return adjusted || sliced;
+  });
+
+  generateColorPalette = _.memoize((color: string): string[] => {
+    const palette = this.generatePalette(color);
     return this.shouldSupportDarkMode && Scheme.getSchemeType() === 'dark' ? _.reverse(palette) : palette;
   });
 
+  generateTokens(color: string, mode: 'light' | 'dark') {
+    return mode === 'light' ? this.generateLightModeTokens(color) : this.generateDarkModeTokens(color);
+  }
+
+  private generateLightModeTokens(color: string) {
```

Comment:
`generateLightModeTokens` and `generateDarkModeTokens` methods are very similar, but I would rather have them separate, then to couple them